test/e2e/scalability/load.go

```diff
@@ -566,8 +566,16 @@ func scaleResource(wg *sync.WaitGroup, config testutils.RunObjectConfig, scaling
 		LabelSelector:   selector.String(),
 		ResourceVersion: "0",
 	}
-	_, err := config.GetClient().CoreV1().Pods(config.GetNamespace()).List(options)
-	framework.ExpectNoError(err, fmt.Sprintf("listing pods from rc %v", config.GetName()))
+	attemptsLeft := 5
```

Comment:
I think we should use exponential backoff here. Otherwise this could just hammer API server to death.
Hopefully the k8s codebase has a library which does this already?

Reply:
We could do that, but I don't really think we need it here. Because this scaleResource function is made to wait for a random amount of time for each RC before doing the scaling (see `sleepUpTo()` call above) which makes sure they're spread across time.

Reply:
I'm not so sure.. for example if the CONNRESET was due to overload, this tight loop will make it worse, by magnifying traffic by a multiplier equal to `attemptsLeft`, potentially causing a cascading failure.

Reply:
Makes sense. How about sleeping for a constant amount of time - like we're doing at other places in e2e framework (e.g https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L817)?

Reply:
That has the downside of needing to tune this constant every now and then. Since we're kind of supposed to work on the technical debt these days, I personally think it would make sense to invest in writing/finding some reusable code for exponential backoff.

Reply:
Changed it to exponential backoff. There is a `wait` pkg in kubernetes already to do this.